client/catapult/src/catapult/crypto/OpensslContexts.cpp

```diff
@@ -37,15 +37,11 @@ namespace catapult { namespace crypto {
 	// region OpensslDigestContext
 
 	OpensslDigestContext::OpensslDigestContext() {
-		std::memset(&m_buffer, 0, CountOf(m_buffer));
-	}
-
-	OpensslDigestContext::~OpensslDigestContext() {
-		reset();
+		m_buffer = UniquePtr(EVP_MD_CTX_new(), [](context_type* p) { EVP_MD_CTX_free(p); });
```

Comment:
this is doing an allocation - that defeats the purpose of these contexts to be stack allocated 😅

Reply:
haha, I was wondering this.  I guess you want stack allocation for performance?

Reply:
yep, would prefer to keep that, if possible